crates/hotshot/src/lib.rs

```diff
@@ -634,7 +634,7 @@ pub trait HotShotType<TYPES: NodeType, I: NodeImplementation<TYPES>> {
 
 #[async_trait]
 impl<
-        TYPES: NodeType,
+        TYPES: NodeType<Transaction = VIDTransaction, BlockType = VIDBlockPayload>,
```

Comment:
This was my concern about making these abstract when we generally only want one implementation. If we are fixing `Transaction` and `BlockType` in library code as general-purpose as this, they probably shouldn't be associated types to begin with. WDYT?

Reply:
I agree that we should not over-generalize. If we only ever expect 1 tx type and 1 block type then just hard-code them. Perhaps an argument could be made that traits are useful so that it's easy to mock things in tests. But I don't like adding complexity to production code merely to facilitate tests.

Regarding associated types: This is a question about idiomatic rust. In this instance we are using assoc types as if they were type params. Presumably that's not idiomatic, but there are reasons why people do it. eg. If there are many type params then the code becomes unreadable---I assume that's why we have a `NodeType` trait to collect all the type params into a single trait so that every single type doesn't need 47 type params. I have not yet formed a strong opinion here.

Reply:
Sorry, I wasn't complaining about these being associated types specifically. I actually don't think they should be generic at all (whether as type parameters or associated types) if we have core library code that doesn't work unless one specific instantiation (`VIDTransaction`, `VIDBlockPayload`) is used. Like, HotShot is generic over `Transaction` and `BlockType` but it seems impossible to use HotShot unless your are using the VID implementations...we should either fix that or just hard code these two specific types.

Reply:
There are two cases where we may want to keep the two traits.
1. We have `DummyBlock` that also implements `BlockPayload`.

2. We may need another `BlockPayload` impl in the future. Pasting the note from an earlier [discussion on Zulip](https://espresso.zulipchat.com/#narrow/stream/311646-Consensus/topic/.E2.9C.94.20use.20of.20VID.20block.20commitment/near/391205141):

> I just had a quick discussion with the consensus team and Ellie mentioned we might need two BlockPayload implementations for TimeBoost (undecided yet, but it's possible), so I'll do the renaming but keep BlockPayload as a trait for more flexibility.
We could convert it into a struct if we figure we only need one implementation later.

For 1, we have a GitHub [issue](https://github.com/EspressoSystems/HotShot/issues/1763) to investigate whether we can remove the dummy struct, and I could do it soon. For 2, can we wait until @elliedavidson is back from vacation, then decide whether we can simplify this?